src/networking/peer.rs

```diff
@@ -480,8 +480,13 @@ impl SaitoPeer {
             }
             "SNDTRANS" => {
```

Comment:
I think there is some missing behaviour here. When we receive the transaction, we need to do something like propagate_transaction_to_peers, except there is some extra logic, for each peer we need to check if they are already in the path, if so, then we do not propagate to that peer.